libs/librrgraph/src/base/rr_spatial_lookup.cpp

```diff
@@ -258,12 +279,40 @@ void RRSpatialLookup::add_node(RRNodeId node,
     rr_node_indices_[type][layer][x][y][side][ptc] = int(size_t(node));
 }
 
+bool RRSpatialLookup::remove_node(RRNodeId node,
+                                  int layer,
+                                  int x,
+                                  int y,
+                                  t_rr_type type,
+                                  int ptc,
+                                  e_side side) {
+    VTR_ASSERT(node.is_valid());
+    VTR_ASSERT_SAFE(4 == rr_node_indices_[type].ndims());
+    VTR_ASSERT_SAFE(layer >= 0);
+    VTR_ASSERT_SAFE(x >= 0);
+    VTR_ASSERT_SAFE(y >= 0);
+    VTR_ASSERT_SAFE(type != NUM_RR_TYPES);
+    VTR_ASSERT_SAFE(ptc >= 0);
+    VTR_ASSERT_SAFE(side != NUM_SIDES);
+
+    if (type >= rr_node_indices_.size()) return false;
+    if ((size_t)layer >= rr_node_indices_[type].dim_size(0)) return false;
+    if ((size_t)x >= rr_node_indices_[type].dim_size(1)) return false;
+    if ((size_t)y >= rr_node_indices_[type].dim_size(2)) return false;
+    if (side >= rr_node_indices_[type].dim_size(3)) return false;
+    if ((size_t)ptc >= rr_node_indices_[type][layer][x][y][side].size()) return false;
+    if (rr_node_indices_[type][layer][x][y][side][ptc] != int(size_t(node))) return false;
```

Comment:
At the moment both casts are needed since ``vtr::StrongId`` has a ``operator size_t()`` but not ``operator int()``. Should I add this operator to ``vtr::StrongId``?

Reply:
Yes, that sound like a good solution. Thanks.